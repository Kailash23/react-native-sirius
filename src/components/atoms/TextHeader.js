import React from 'react';
import { View } from 'react-native';
import { customFonts } from '@styles/customFonts';
import { withTheme, Divider } from 'react-native-paper';
import { SIZES } from '@styles/sizes';
import STitle from '@atoms/STitle';
import { layout } from '@utils/Layout';

function TextHeader(props) {
  const { headerTitle, theme } = props;
  return (
    <View style={{ backgroundColor: theme.colors.background }}>
      <STitle
        avoidTextScaling
        fontFamily={customFonts.torstarDecSemiBold}
        title={headerTitle}
        size={SIZES.BASE * 2}
        opacity={theme.dark ? 1 : 0.78}
        style={styles.title}
      />
      <Divider style={[styles.divider, { borderColor: theme.colors.line }]} />
    </View>
  );
}

export default withTheme(TextHeader);

const styles = {
  title: {
    paddingHorizontal: SIZES.PADDING + 4,
    letterSpacing: 0.4,
    paddingTop: layout.isMobile ? 20 : SIZES.BASE * 3,
    paddingBottom: 5
  },
  divider: {
    borderWidth: 0.5,
    marginHorizontal: SIZES.BASE
  }
};
