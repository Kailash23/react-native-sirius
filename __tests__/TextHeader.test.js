import TextHeader from '@atoms/TextHeader';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({ adapter: new Adapter() });

describe('Testing <TextHeader/>', () => {
  test('should render a TextHeader', () => {
    const wrapper = shallow(<TextHeader headerTitle={'Sample Header title'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
