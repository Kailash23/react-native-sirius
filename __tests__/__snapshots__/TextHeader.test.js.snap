// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TextHeader/> should render a TextHeader 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  headerTitle="Sample Header title"
/>
`;
